### PR TITLE
Keep only two routers, one for docker and another for crio/containerd

### PR DESCRIPTION
**Pull request by VihasMakwana**
See https://github.com/open-telemetry/opentelemetry-helm-charts/issues/819.

We can use two routers as, crio and containerd are similar
The recombine operator is same for both crio and containerd, so we can use just one.

This also fixes an probable issue https://github.com/signalfx/splunk-otel-collector-chart/issues/644

### CODE REVIEW

**Comment by Allex1** [comment]
@VihasMakwana  thanks for your contribution .Please follow our [guide](https://github.com/open-telemetry/opentelemetry-helm-charts/blob/main/charts/opentelemetry-collector/CONTRIBUTING.md) in order for the ci to pass.

**Review by TylerHelmuth — COMMENTED** [review]
@VihasMakwana can please share how you tested this change to be equivalent to the current solution of using 2 routers?  @djaglowski can you share your thoughts on the changes to the filelogreceiver configuration?

**Comment by VihasMakwana** [comment]
So, for this, I have tested it the following way:

- For docker-based containers, I have tested them using minikube and splunk_hec to verify the timestamps and event delivery
- For crio and containerd-based containers, I have tested them using minikube and microk8s, and splunk_hec to verify the timestamps and event delivery and covered the following scenarios:

     -  Used zulu timestamp and verified that it was being received with no errors
     -  Used timestamp with local offset (rfc3393 spec) and verified that it was being received with no errors

**Review by djaglowski — COMMENTED** [review]
I agree that the recombine operators for crio vs containerd are redundant and can be consolidated, but I do not understand the other changes.

At a glance the regex changes do not look correct to me. Can you give some specific examples that demonstrate how `"^[^ Z]+ "` and `"^[^ Z]+Z"` are correctly combined into `"^[^ ]+ "`?

**Comment by VihasMakwana** [comment]
@djaglowski please check https://github.com/signalfx/splunk-otel-collector-chart/issues/644. 
I made changes to regex because it was hard-coded to accept zulu.

**Comment by VihasMakwana** [comment]
`"^[^ ]+ "` matches the timestamp. 
It will match `2023-01-23T12:09:28.344779078-06:00` and also, `2023-07-06T13:08:23.702Z`. 
`"^[^ Z]+Z"` only matches the zulu timestamp and `"^[^ Z]+ "` only matches non-zulu i.e. with timezone offset.

File: charts/opentelemetry-collector/examples/daemonset-collector-logs/rendered/configmap-agent.yaml [hunk 1]
```diff
@@ -36,35 +36,19 @@ data:
           routes:
           - expr: body matches "^\\{"
             output: parser-docker
-          - expr: body matches "^[^ Z]+ "
-            output: parser-crio
-          - expr: body matches "^[^ Z]+Z"
-            output: parser-containerd
+          - expr: body matches "^[^ ]+ "
+            output: parser-crio-containerd
```
[the remark this answers is not in the document]
Comment (VihasMakwana) [reply charts/opentelemetry-collector/examples/daemonset-collector-logs/rendered/configmap-agent.yaml:40]: Yeah.

**Comment by djaglowski** [comment]
Thanks for explaining @VihasMakwana. I think your explanation of the regexes makes sense but I haven't tested them so I'd appreciate others to look at this as well.

Personally I find it confusing that there is a bug fix in the same PR as non-functional changes. Both changes would be simpler to understand if presented in isolation. Perhaps you'd consider splitting the PR.

Finally, if these changes are accepted here, would you please submitting equivalent PRs to the contrib example [here](https://github.com/open-telemetry/opentelemetry-collector-contrib/tree/main/examples/kubernetes)?

**Comment by VihasMakwana** [comment]
Makes sense.
But the issue is about keeping two routers, and for that to accomplish the changes in regex is required. And that also fixes the issue I attached in previous comment.

**Comment by VihasMakwana** [comment]
and yes, will submit PRs to contrib.

**Comment by VihasMakwana** [comment]
Hmm, I have an open PR for the similar thing in splunk repo, maybe we can wait until that one gets accepted and tested?

**Comment by VihasMakwana** [comment]
^^

**Comment by VihasMakwana** [comment]
@atoulme done

**Comment by TylerHelmuth** [comment]
@VihasMakwana have you had a chance to release this in the Splunk helm chart and confirm it is equivalent?

**Comment by VihasMakwana** [comment]
@TylerHelmuth No, not yet. If you or anyone else can test this out on your end, it would be of great help ;).

**Comment by VihasMakwana** [comment]
@atoulme can you re-review this one?

**Pull request closed without merging**